addons/survey/controllers/main.py

```diff
@@ -535,11 +542,23 @@ def survey_submit(self, survey_token, answer_token, **post):
             answer_sudo.write({'last_displayed_page_id': post['previous_page_id']})
             # Go back to specific page using the breadcrumb. Lines are saved and survey continues
             return self._prepare_question_html(survey_sudo, answer_sudo, **post)
+        elif 'next_skipped_page_or_question' in post:
+            answer_sudo.write({'last_displayed_page_id': page_or_question_id})
+            return self._prepare_question_html(survey_sudo, answer_sudo, **{
+                'next_skipped_page': True
+            })
         else:
             if not answer_sudo.is_session_answer:
                 next_page = survey_sudo._get_next_page_or_question(answer_sudo, page_or_question_id)
                 if not next_page:
-                    answer_sudo._mark_done()
+                    if survey_sudo.users_can_go_back and answer_sudo.user_input_line_ids.filtered(
+                            lambda a: a.skipped and a.question_id.constr_mandatory):
+                        answer_sudo.write({'last_displayed_page_id': page_or_question_id})
```

Comment:
Can't we do that at the start of the block to avoid having the same line twice? I don't think it affects any of the computations and it's done in all cases

Reply:
Where would you put it exactly ? Actually it's not called when `_mark_done` is and it's not relevant to do so even if it probably doesn't matter